File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.java
class Solution {
    public int solution(int n, int w, int num) {
        int cur = (num - 1) % w;
        int left = (n - num) / (2 * w), right = (n - num) % (2 * w);
        return 1 + 2 * left + (right >= 2 * (w - 1 - cur) + 1 ? 1 : 0);
    }
}

File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.py
def solution(n, w, num):
    num_col = (num - 1) % w
    left, right = divmod(n - num, 2 * w)
    return 1 + 2 * left + (1 if right >= 2 * (w - 1 - num_col) + 1 else 0)